infra/gcp/prow/ensure-e2e-projects.sh

```diff
@@ -165,24 +165,35 @@ for prj; do
 
     color 6 "Ensure prow-build prowjobs are able to ssh to instances in e2e project: ${prj}"
     prow_build_ssh_pubkey="prow:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCmYxHh/wwcV0P1aChuFLpl28w6DFyc7G5Xrw1F8wH1Re9AdxyemM2bTZ/PhsP3u9VDnNbyOw3UN00VFdumkFLjLf1WQ7Q6rZDlPjlw7urBIvAMqUecY6ae1znqsZ0dMBxOuPXHznlnjLjM5b7O7q5WsQMCA9Szbmz6DsuSyCuX0It2osBTN+8P/Fa6BNh3W8AF60M7L8/aUzLfbXVS2LIQKAHHD8CWqvXhLPuTJ03iSwFvgtAK1/J2XJwUP+OzAFrxj6A9LW5ZZgk3R3kRKr0xT/L7hga41rB1qy8Uz+Xr/PTVMNGW+nmU4bPgFchCK0JBK7B12ZcdVVFUEdpaAiKZ prow"
+    ssh_keys_expected=(
+      "${prow_build_ssh_pubkey}"
+      # TODO(amwat,spiffxp): something is adding an extra prow: prefix, it is
+      # unclear where in prow->kubetest2->cluster/log-dump.sh->`gcloud ssh`
+      # this is happening
+      "prow:${prow_build_ssh_pubkey}"
+    )
 
     # append to project-wide ssh-keys metadata if not present
     ssh_keys_before="${TMPDIR}/ssh-keys.before.txt"
     ssh_keys_after="${TMPDIR}/ssh-keys.after.txt"
     gcloud compute project-info describe --project="${prj}" \
       --format='value(commonInstanceMetadata.items.filter(key:ssh-keys).extract(value).flatten())' \
-      | sed '$d' > "${ssh_keys_before}"
+      | sed -e '/^$/d' > "${ssh_keys_before}"
 
     cp "${ssh_keys_before}" "${ssh_keys_after}"
 
-    if [ "${K8S_INFRA_ENSURE_E2E_PROJECTS_RESETS_SSH_KEYS}" == "true" ]; then
-      echo "${prow_build_ssh_pubkey}" > "${ssh_keys_after}"
-    elif ! grep -q "${prow_build_ssh_pubkey}" "${ssh_keys_before}"; then
-      echo "${prow_build_ssh_pubkey}" >> "${ssh_keys_after}"
+    if [ "${K8S_INFRA_ENSURE_E2E_PROJECTS_RESETS_SSH_KEYS:-"false"}" == "true" ]; then
+      printf '%s\n' "${ssh_keys_expected[@]}" > "${ssh_keys_after}"
+    else
+      for ssh_key in "${ssh_keys_expected[@]}"; do
+        if ! grep -q "${ssh_key}" "${ssh_keys_before}"; then
+          echo "${ssh_key}" >> "${ssh_keys_after}"
+        fi
+      done
     fi
 
     if ! diff ${ssh_keys_before} ${ssh_keys_after} >/dev/null; then
-      gcloud compute project-info add-metadata --project="${prj}" \
+      cloud compute project-info add-metadata --project="${prj}" \
```

Comment:
@spiffxp Nit?

Reply:
good catch, thank you

Reply:
https://github.com/kubernetes/k8s.io/pull/1907 should fix